### PR TITLE
Add test documentation

### PR DESCRIPTION
**Pull request by atodorov**
This is the start of what is supposed to be a human readable test documentation of the Anaconda test suite. Filing here for comments. 

I'd like to also include API docs for the Python files under tests/ but so far I'm having troubles building these. I will likely have to add __init__.py files to all directories that are missing them and fix the import failures from Sphinx (e.g. by altering PYTHONPATH if necessary).

Btw it looks like the generated HTML is under version control, not sure if that needs to be removed.

### CODE REVIEW

**Comment by jkonecny12** [comment]
I like this, good work.
Seems good to me but this will be better if someone who can check grammar and have better understanding of the tests look on it too.

**Comment by atodorov** [comment]
Can you comment on the inclusion of test API docs and the changes required to do so? Is this acceptable ?

**Comment by jkonecny12** [comment]
I fear I can't help you much with this. I think this question is more for @bcl.

**Comment by atodorov** [comment]
Update:

* The added init files are required for Sphinx to be able to discover all tests. They don't affect test results;
* The config changes in docs/conf.py are required so that we can import all tests without failures except pylint
```
************* Module ../docs/conf.py
W9940(environment-modify): 28,4: : Found potentially unsafe modification of environment
************* Module ../tests/gui/inside/__init__.py
/usr/lib/python3.4/site-packages/astroid/scoped_nodes.py:425: PyGIWarning: Atspi was imported without specifying a version first. Use gi.require_version('Atspi', '2.0') before import to ensure that the right version gets loaded.
  return MANAGER.ast_from_module_name(absmodname)
************* Unused False Positives Found:
^F0401.*: Unable to import 'dogtail.*'$
FAIL pylint/runpylint.py (exit status: 4)
```
* Currently the building of the docs fails because of Dogtail for Python3
  - it is still not officially available in Rawhide (I've tested with a manually build package)
  - it fails because assistive technologies are disabled. Even with a patch to skip aborting it still fails because I don't have at-spi-bus-launcher running.

File: docs/conf.py [hunk 1]
```diff
@@ -21,6 +21,12 @@
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 sys.path.insert(0, os.path.abspath('..'))
 
+# configuration required to import test modules
+for path in ["../pyanaconda/isys/.libs", "../pyanaconda", "../tests", "../tests/lib", "../dracut", "../widgets"]:
```
Comment (vpodzime) [inline docs/conf.py]: I'm not sure if these "raw" (non-prefixed)  relative paths are a good way to go. Or do they work the same no matter which directory you run the sphinx stuff from?

**Comment by vpodzime** [comment]
Looks good to me otherwise.

File: docs/conf.py (hunk 1 above)
Comment (atodorov) [inline docs/conf.py]: Works even from the top level directory.

**Comment by bcl** [comment]
Yes, including the generated docs is on purpose. I do a separate run to generate them and copy them over to the gh-pages branch so they'll show up on github. I'm all for documenting the test procedures, but I'm not sure how useful it is to actually document the test code with sphinx. It generally isn't very complicated and if it is going to cause problems with needing packages like dogtail and at-spi-bus-launcher it isn't worth it to block generating the rest of the docs.

**Comment by atodorov** [comment]
@bcl - I'm removing the gui/ tests documentation for now b/c of the issues with Dogtail and b/c it's not clear if these tests will be available much longer.

**Comment by bcl** [comment]
> Btw I see the following issues if I don't have X running:

Yeah, GTK doesn't really like sphinx on a non X system. I build the docs with my dev system and the helper script passes $DISPLAY through so that it doesn't complain.

**Comment by atodorov** [comment]
@bcl - fixed and slightly updated the docs. Please review and merge if no objections.

**Comment by atodorov** [comment]
ping, any update on this one ?

**Comment by vpodzime** [comment]
Still looks good to me.

**Comment by bcl** [comment]
Thanks, tested and added some new modules to make-sphinx-docs.

**Pull request closed without merging**